colony-common/src/lib.rs

```diff
@@ -0,0 +1,143 @@
+    NotEnoughBalance,
+    /// When the account sequence given in the transaction is invalid.
+    #[error("invalid account sequence; expected {0} but got {1}")]
+    InvalidAccountSequence(u64, u64),
+    /// When the contract sequence given in the transaction is invalid.
+    #[error("invalid account sequence; expected {0} but got {1}")]
+    InvalidContractSequence(u64, u64),
+    /// When the contract fails to execute the transaction with its own error.
+    #[error("internal contract error: {0}")]
+    InternalContractError(String),
+    /// When the contract is missing.
+    #[error("no such contract: {0}")]
+    NoSuchContract(String),
+    /// An unknown error.
+    #[error("unknown error: {0}")]
+    Unknown(String),
+}
+
+/// An abstraction of a colony chain, which is viewed by the relayer and the explorer.
+///
+/// Every colony chains MUST have at least two types of contracts: the light client and the treasury.
+/// This interface directly specifies the actions applicable to the two essential contracts,
+/// whereas it is generalized as an opaque packet for those non-essential, chain-local and application-specific contracts.
+///
+/// One trivial implementation of this trait would carry the address of the full node and
+/// a relayer account used to submit message delivering transactions.
+#[async_trait]
+pub trait ColonyChain {
+    /// Returns the name of the chain.
+    fn get_chain_name(&self) -> String;
+
+    /// Checks whether the chain is healthy and the full node is running.
+    async fn check_connection(&self) -> Result<(), Error>;
+
+    /// Returns the list of the contracts that are available on the chain
+    async fn get_contract_list(&self) -> Result<Vec<ContractInfo>, Error>;
+
+    /// Returns the address and the current balance (which is used to pay the gas fee) of the relayer account in this chain.
+    ///
+    /// Note that there is no authority of the relayer account; it is just a convenient account to pay the gas fee
+    /// (i.e., there is no need to check the transaction submitter by the contract).
+    async fn get_relayer_account_info(&self) -> Result<(String, Decimal), Error>;
+
+    /// Returns the latest header that the light client has verified.
+    async fn get_light_client_header(&self) -> Result<Header, Error>;
+
+    /// Returns the current balance of fungible tokens in the treasury contract.
+    async fn get_treasury_fungible_token_balance(&self) -> Result<HashMap<String, Decimal>, Error>;
+
+    /// Returns the current balance of non-fungible tokens in the treasury contract, identified as `(collection address, collection index(chain-specific))`.
+    async fn get_treasury_non_fungible_token_balance(&self)
+        -> Result<Vec<(String, String)>, Error>;
+
+    /// Updates the light client state by providing the next, valid block header and its proof.
+    ///
+    /// A transaction that carries the given data will be submitted to the chain.
+    async fn deliver_message_update_light_client(
```

Comment:
```suggestion
    async fn deliver_message_update_light_client(
    async fn update_light_client_with_message(
```

주석과 함수명을 통일성있게 유지하는게 좋을 것 같습니다. `deliever` 가 앞에 있으니까 이름만 보고 뭐하는 함수인지 직관적으로 안 와닿고 주석을 보니 더 헷갈렸습니다. 메세지가 중요한게 아니라면 뒤에 메세지도 빼도 되지 않을까 싶습니다 update 와 순서를 바꾸고. 만약 메세지가 더 중요한 것이라면, 주석을 `Updates ~~~` 가 아니라 `Deliver message about updating light client ~~` 로 가야할 것 같습니다. 

하기 함수들도 마찬가지

Reply:
fixed